### PR TITLE
b2c for provider auth

### PR DESCRIPTION
**Pull request by kristinmerbach**
Switches the auth used for the provider gateway to use azure b2c. also adds new envs to deploys in GHAs

### CODE REVIEW

File: apps/provider-gateway/src/app/app.module.ts [hunk 1]
```diff
@@ -1,56 +1,164 @@
...
 import { APP_TITLE } from '@dbh/theme';
 import { RootStoreModule } from '@dbh/shared/state/root-store';
+import { environment } from '../environments/environment';
+
 import { AppComponent } from './app.component';
 
+const subdomain = environment.B2C_SUBDOMAIN || '';
+const gatewayCid = environment.NX_GATEWAY_C_ID || '';
+
+const isIE =
```
Comment (bbodine1) [inline apps/provider-gateway/src/app/app.module.ts:41]: I don't think we need this check. We won't support IE.

File: apps/provider-gateway/src/app/app.module.ts [hunk 2]
```diff
@@ -1,56 +1,164 @@
...
-          canLoad: [AuthGuard],
         },
         {
           path: '**',
           redirectTo: 'provider-gateway',
           pathMatch: 'full',
         },
       ],
-      { initialNavigation: 'enabledBlocking' }
+      {
```
Comment (bbodine1) [inline apps/provider-gateway/src/app/app.module.ts:116]: Is there a reason we are checking if we are loading in an iFrame?

**Review by bbodine1 — APPROVED** [review]

File: apps/provider-gateway/src/app/app.component.ts [hunk 3]
```diff
@@ -7,4 +16,47 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
```
[the remark this answers is not in the document]
Comment (kristinmerbach) [reply apps/provider-gateway/src/app/app.component.ts]: Done

File: apps/provider-gateway/src/app/app.module.ts (hunk 1 above)
Comment (kristinmerbach) [reply apps/provider-gateway/src/app/app.module.ts:41]: Will do at a latter point

File: apps/provider-gateway/src/app/app.module.ts (hunk 2 above)
Comment (kristinmerbach) [reply apps/provider-gateway/src/app/app.module.ts:116]: It is due to it being a msal option

**Pull request merged**